enterprise/server/usage/usage.go

```diff
@@ -273,28 +273,80 @@ func (ut *tracker) flushCounts(ctx context.Context, groupID string, c collection
 	}
 	dbh := ut.env.GetDBHandle()
 	return dbh.TransactionWithOptions(ctx, db.Opts().WithQueryName("upsert_usage"), func(tx *db.DB) error {
+		// First check whether the row already exists. Make sure to select for
+		// update in order to lock the row.
+		rows, err := tx.Raw(`
+			SELECT *
+			FROM "Usages"
+			WHERE
+				region = ?
+				AND group_id = ?
+				AND period_start_usec = ?
+			`+dbh.SelectForUpdateModifier(),
+			pk.Region,
+			pk.GroupID,
+			pk.PeriodStartUsec,
+		).Rows()
+		if err != nil {
+			return status.WrapError(err, "query rows")
+		}
+
+		schema, err := db.TableSchema(tx, &tables.Usage{})
+		if err != nil {
+			return status.WrapError(err, "schema")
+		}
+
+		existingRowCount := 0
+		for rows.Next() {
+			existingRowCount++
+			fields := map[string]any{}
+			if err := tx.ScanRows(rows, &fields); err != nil {
+				return status.WrapError(err, "scan row")
```

Comment:
I don't think the WrapErrors are adding much value, can we remove these?

Reply:
I'll remove in this case since DB errors are usually more obvious, but I've sort of started to form a habit of adding these because I've gotten bitten lately by error messages in the logs being hard to decipher, e.g. https://github.com/buildbuddy-io/buildbuddy/pull/4342

When this happens it's a bit unfortunate because it adds another release cycle before we can be sure about where an error is coming from.